### ISSUE TITLE
Error loading Python lib when opening exec file(OS X)

### ISSUE THREAD

**Issue by seraph0012** [pyinstaller/pyinstaller#2150]
I keep getting the same error message when opening the exec file(No error in the build phase), here is the error message:

Error loading Python lib '/Users/apple/Desktop/PythonFIles/kivy/ric/RICApp/build/ric/Python': dlopen(/Users/apple/Desktop/PythonFIles/kivy/ric/RICApp/build/ric/Python, 10): image not found

I've tried different options from simplest 'pyinstaller main.py' to some complex ones like

pyinstaller -y --clean --windowed --name touchtracer   --exclude-module _tkinter   --exclude-module Tkinter   --exclude-module enchant   --exclude-module twisted main.spec

I also tried to edit the spec file when the default spec file doesn't work, but I just got the same error on opening the exec file.

The error message is simple but I've no idea where the error comes from. I don't think the error is suggesting that I'm missing any dependencies or modules since the lib it fails to load is just Python. My main.py is just print('hello world') so I think the error doesn't come from my code. 

I've tried reinstall using pip/directly clone from git. I've tried both the 3.2 version and the dev versions. I've googled the error message and find there's one similar issue on the git repo. The issue has the same error message but it's on a very old version of pyinstaller and one of the replies says that the issue has already been fixed. I can't find any other useful issues or stackoverflow questions related to this error.

My OS X version: OS X EI Capitan 10.11.5
python: 2.7.10
pip: 8.1.2
pyinstaller: 3.2/dev(tried both, get the same error)

I think I've reinstalled all the related dependencies several times today and they should all be updated to the latest version.

**Comment by htgoebel** [comment]
This looks like not all required files are packages and thus some image is missing. Please have a look at [Make sure everything is packaged correctly](https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs#make-sure-everything-is-packaged-correctly) for tips to find out what is missing.

**Comment by seraph0012** [comment]
> This looks like not all required files are packages and thus some image is missing. Please have a look at Make sure everything is packaged correctly for tips to find out what is missing.

I only have one file main.py and the content is just print('hello world'). That's all I want to package. I didn't import any modules or require any resources in the main.py file. 
I didn't get any ImportErrors or File not found errors. It looks like the error message suggests that the file/resource I'm missing is Python, which I definitely have installed on my computer.

**Comment by tallforasmurf** [comment]
The file `/Users/apple/Desktop/PythonFIles/kivy/ric/RICApp/build/ric/Python` would perhaps be the Python interpreter? In a quite nonstandard location? So how was it built? Is there a static lib for PyInstaller to include? Because part of the packaging process is to bundle the Python interpreter as a static lib into the bundled app. It looks for that static lib based on the Python that is executing the PyInstaller command. Could you show us the error messages from the actual PyInstaller command execution? I suspect is has a "not found" for Python.

**Comment by seraph0012** [comment]
> The file /Users/apple/Desktop/PythonFIles/kivy/ric/RICApp/build/ric/Python would perhaps be the Python interpreter? In a quite nonstandard location? So how was it built?

In the building phase it seems that pyinstaller tried to find the python binary file:

1462 INFO: Python library not in binary depedencies. Doing additional searching...
1464 INFO: Using Python library /System/Library/Frameworks/Python.framework/Versions/2.7/Python

I think that's where the error comes from. My python binary file is actually located at '/usr/bin/python' and it's in the PATH environment variable. Do I need to add any new environment variable so that pyinstaller can find my python binary without specifying it in spec file or option flags? I don't see any instructions on linking python binary file under OS X in official docs.

I also tried to add '/usr/bin/python' to 'binaries' in 'Analysis' in the spec file, although the official doc says the 'binaries' field is for 'non-python modules needed by the scripts'. Then I got 'ValueError: too many values to unpack' after adding the python binary in 'biinaries'

Instead of just run 'pyinstaller main.py' or 'pyinstaller main.spec', I tried to run it with python:
python -m PyInstaller main.py
That doesn't work either.
